otter/models/interface.py

```diff
@@ -614,92 +614,36 @@ def delete_webhook(policy_id, webhook_id):
         """
 
 
-class NoSuchServerIntentError(Exception):
+class IScalingGroupServersCache(Interface):
     """
-    Error to be raised when attempting operations on a server intent that does not
-    exist.
-    """
-    def __init__(self, tenant_id, group_id, server_id):
-        super(NoSuchServerIntentError, self).__init__(
-            "No such server {s} in group {g} for tenant {t}"
-            .format(t=tenant_id, g=group_id, s=server_id))
-
-
-class IScalingGroupServerIntentsCollection(Interface):
+    Cache of servers in scaling groups
     """
-    Collection of servers intended to be there in a scaling group. Each server in the
-    this group should eventually match to a real server in Nova. All operations on this
-    model will not have any impact on real Nova servers. It is the caller's responsibility
-    to sync them (if needed).
-    """
-
-    def create_server_intent(log, status='pending'):
-        """
-        Create server intended to be there in scaling group
-
-        :param :class:`BoundLog` log: A bound logger
-        :param str status: status of the server. one of 'pending' or 'active'
-
-        :return: a :class:`twisted.internet.defer.Deferred` that fires with ``dict``
-                corresponding with :data:`otter.json_schema.model_schemas.server`
-        :raises NoSuchScalingGroupError: if this scaling group does not exist
-        """
-
-    def update_server_intent(log, server_intent_id, nova_id, status, lb_info):
-        """
-        Update existing server intent information
-
-        :param :class:`BoundLog` log: A bound logger
-        :param str server_intent_id: ID of server intent
-        :param str nova_id: Server ID of corresponding Nova instance
-        :param str status: server status. One of 'pending' or 'active'
-        :param `dict` lb_info: Load balancer information dict. This will be stored as JSON
-
-        :return: a :class:`twisted.internet.defer.Deferred` that fires with None
+    tenant_id = Attribute("Rackspace Tenant ID of the owner of this group.")
+    group_id = Attribute("UUID of the scaling group - immutable.")
 
-        :raises NoSuchScalingGroupError: if this scaling group does not exist
-        :raises NoSuchServerIntentError: if the server intent id does not exist
+    def get_servers():
         """
-
-    def list_server_intents(log, status=None, limit=100, marker=None):
+        Return latest cache of servers in a group along with last time the
+        cache was updated.
+        :return: Effect of (servers, last update time) tuple where servers
+            is list of dict and last update time is datetime object. Will
```

Comment:
That happens to be done due to the way the cache needs to be merged with current servers. The servers after that are list of dicts. Also, the insertion of servers are as list of dicts. I think its fine to leave it like this :)